tooling/ramdisk.md

```diff
@@ -0,0 +1,81 @@
+# RAMdisk
+You can mount some of your computer's memory to your filesystem in order to perform read/write intensive tasks very, very quickly such as processing large JSON files, or to [work with sensitive documents](./encryption.md) that you don't want to risk writing to disk. This is called a RAM disk.
+
+<!-- contents box begin -->
+<table>
+<tr/>
+<tr>
+<td>
+<p/>
+<div align="center">
+<b>Contents</b>
+</div>
+<p/>
+<!-- contents markdown begin -->
+
+1. [Preparation](#preparation)
+1. [Creation](#creation)
+1. [Destruction](#destruction)
+1. [See Also](#see-also)
+
+<!-- contents markdown end -->
+<p/>
+</td>
+</tr>
+</table>
+<!-- contents box end -->
+
+## Preparation
+Make a folder to mount the RAM disk to.
+```bash
+sudo mkdir -p /mnt/ram
+```
+Take ownership of that folder.
+```bash
+sudo chown "$USER:$USER" /mnt/ram
+```
+Put a file there so you can tell whether or not a RAM disk is mounted.
+```bash
+echo 'WARNING: If you can see this, there is currently no RAM disk mounted!' > /mnt/ram/STOP.txt
+```
```

Comment:
Alternatively what I like to do is remove permissions from the empty directory that will be the mount point. That will prevent reading or writing or even looking in to the directory until something is mounted there

Reply:
But then how do you know the mount point exists?

Can you please provide an example, if you want me to add it?

Reply:
You don't need to add it.

Basically something like
```
sudo mkdir /mnt/ram
sudo chmod 0600 /mnt/ram
```
Now it's impossible (non-root) to `ls /mnt/ram` or `touch /mnt/ram/foobar`. So you can't accidentally use it when there is nothing mounted there. But it's still possible to mount something _at_ `/mnt/ram`.